examples/trainer_keras.py

```diff
@@ -28,83 +33,124 @@ def ensure_results_dir():
+        x_test, e_test, y_test = get_m4_data_multivariate(model.backcast_length, model.forecast_length, is_training=False)
+    elif task == 'kcg':
+        x_test, e_test, y_test = get_kcg_data(model.backcast_length, model.forecast_length, is_training=False)
+    elif task == 'nrj':
+        x_test, e_test, y_test = get_nrj_data(model.backcast_length, model.forecast_length, is_training=False)
     else:
         raise Exception('Unknown task.')
 
-    # x_test, y_test = get_m4_data(model.backcast_length, model.forecast_length, is_training=False)
-    print('x_test.shape=', x_test.shape)
+    print('x_test.shape=', x_test.shape)  
 
     for step in range(model.steps):
         if task == 'm4':
-            x_train, y_train = get_m4_data(model.backcast_length, model.forecast_length, is_training=True)
-        model.train_on_batch(x_train, y_train)
+            x_train, e_train, y_train = get_m4_data_multivariate(model.backcast_length, model.forecast_length, is_training=True)
+        if task == 'kcg':
+            x_train, e_train, y_train = get_kcg_data(model.backcast_length, model.forecast_length, is_training=True)
+        if task == 'nrj':
+            x_train, e_train, y_train = get_nrj_data(model.backcast_length, model.forecast_length, is_training=True)            
+        
+        if model.has_exog():
+            model.train_on_batch([x_train, e_train], y_train)
+        else:
+            model.train_on_batch(x_train, y_train)
+        
         if step % model.plot_results == 0:
             print('step=', step)
             model.save('results/n_beats_model_' + str(step) + '.h5')
-            predictions = model.predict(x_train)
-            validation_predictions = model.predict(x_test)
+            if model.exo_dim > 0:
+                predictions = model.predict([x_train, e_train])
+                validation_predictions = model.predict([x_test, e_test])
+            else:
+                predictions = model.predict(x_train)
+                validation_predictions = model.predict(x_test)
             smape = get_metrics(y_test, validation_predictions)[0]
             print('smape=', smape)
             if smape < model.best_perf:
                 model.best_perf = smape
                 model.save('results/n_beats_model_ongoing.h5')
-            plot_keras_model_predictions(model, False, step, x_train[0, :], y_train[0, :], predictions[0, :])
-            plot_keras_model_predictions(model, True, step, x_test[0, :], y_test[0, :], validation_predictions[0])
+            for l in range(model.input_dim):
+                plot_keras_model_predictions(model, False, step, x_train[0, :, l], y_train[0, :, l], predictions[0, :, l], axis=l)
+                plot_keras_model_predictions(model, True, step, x_test[0, :, l], y_test[0, :, l], validation_predictions[0, :, l], axis=l)
 
-    model.save('results/n_beats_model.h5')
+    model.nbeats.save('results/n_beats_model.h5')
```

Comment:
model.save('results/n_beats_model.h5') is enough here I guess.

Reply:
Sorry !